_docs/ideal_indirection.md

```diff
@@ -14,32 +14,52 @@ wikibook:
 Each component of virtual memory has been split up into several files to help you build a mental model of virtual memory.
-Reading through these files can start to help you understand the roles of the different hardware and software involved in managing virtual memory.
-You will only have to write two functions in `mmu.c`, but it requires a good understanding of the coursebook and decent knowledge of the rest of the provided code.
+Reading through these files will help you understand the roles of the different hardware and software involved in managing virtual memory.
+You will only have to write two functions in `mmu.c`, but it requires a good understanding of the virtual memory translation process and decent understanding of the rest of the provided code.
+
+## How To Tackle This Assignment
+
+* First, you will need to understand how the virtual memory translation process works. You will be writing functions to translate virtual addresses into physical addresses, so a good first step is to understand that process. The coursebook has details on this process, and feel free to ask about this if you are unsure.
+* Then, you will want to understand how we simulate the different components of a computer in this assignment. The header files describe the various structs involved in this simulation, as well as their member variables. Understand them well, and take note of the various functions you can use (you will not be using all of them). The source code for these functions have been given to you. You should be able to get a grasp on the whole model just by reading the header files, but if you are interested to see how we model the different components, you are free to read the source code.
+* Once you have a good mental model, you will want to plan out the steps your code needs to perform. Start by listing the steps involved in translating virtual addresses to physical addresses. Then, add in the steps where you would perform other tasks like
+  * updating the bits of the table entries
+  * checking for tlb caches
+  * performing a context switch
+  * raising page faults and segfaults
+  * requesting memory frames from the kernel and retrieving swap files from the disk
+  * other stuff
+* Once this is done, turn these steps into actual code. Then, test with the given test cases.
 
 The rest of this documentation will serve to help you understand the purpose of each file and give you a high level understanding of virtual memory.
-The implementation details of each function are documented in the header and source files.
-It is your responsibility to read every line of code before starting this assignment, since you will not be able to do anything meaningful otherwise.
+The implementation details of each function are documented in the header files.
+The recommended sequence of reading the provided files will follow the sequence of files introduced in this documentation.
```

Comment:
I hope students follow this...for their own good...

Reply:
I hope so too. Anyone who jumps straight into `mmu.h` without reading the other files will have a headache